docs-sources/includes/_Events.md

```diff
@@ -1,8 +1,56 @@
 # Events
 
-We have seen in the [States](#states) section of this user guide that
-states can be used to emit events between players. Let's dig a bit deeper
-into how this can be used.
+We have seen in the [States](#states) section of this user guide that states can be used to emit events between players.
+Before explaining how to send events, we will first see how MAGE send events to the clients, and how to configure the server to do so.
+
+## How does it work ?
+
+Events sending and receipt is done via **Message Stream**.
+
+**Message Stream** is a protocol used by MAGE servers and its clients in order to communicate. This protocol is implemented through different transports, which are the following:
+
+- **short-polling**: The client send a request to the server and get an **instant response** with the events. The client repeats this every X seconds to receive new events.
+- **long-polling**: The client send a request to the server and keep the connection open. The server will send a response with the events once one or multiple events destinated to the client are received. Then, the client open a new connection and repeat the process.
+- **websocket**: Real time events sending and receipt.
+
+If you need more information about **Message Stream** protocol, you can read this [documentation](https://github.com/mage/mage/tree/master/lib/msgServer/msgStream).
+
+In your MAGE config file, you can specify the priority of the transports which will be used by the client SDK to receive the events.
+
+> config/default.yaml
+
+```yaml
+server:
+    msgStream:
+        detect:
+            - websocket
+            - longpolling
+            - shortpolling
+
+```
+
+You can also configure each transport:
+
+> config/default.yaml
+
+```yaml
+server:
+    msgStream:
+        transports:
+            longpolling:
+                route: "/msgstream"
+                heartbeat: 60
+            shortpolling:
```

Comment:
I think shortpolling also uses a heartbeat.

Reply:
shortpolling can use a heartbeat, but won't it become a longpolling then ?

Reply:
heartbeat would refer to the frequency at which the client should poll I think... but might be an sdk config thing, not server... so forget what I said :)